pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -151,153 +124,71 @@ public SegmentPreprocessingJob(final Properties properties) {
-    }
-
-    if (_enableSorting) {
-      fetchSortingConfig();
-      _logger.info("{}: {}", SORTED_COLUMN_CONFIG, _sortedColumn);
-    }
+    Path sampleAvroPath = inputDataPaths.get(0);
+    int numInputPaths = inputDataPaths.size();
 
-    if (_enableResizing) {
-      fetchResizingConfig();
-      _logger.info("minimum number of output files: {}", _numberOfOutputFiles);
-    }
-
-    job.getConfiguration().set(JobContext.JOB_NAME, this.getClass().getName());
-    // Turn this on to always firstly use class paths that user specifies.
-    job.getConfiguration().set(MRJobConfig.MAPREDUCE_JOB_USER_CLASSPATH_FIRST, "true");
-    // Turn this off since we don't need an empty file in the output directory
-    job.getConfiguration().set(FileOutputCommitter.SUCCESSFUL_JOB_OUTPUT_DIR_MARKER, "false");
-
-    job.setJarByClass(SegmentPreprocessingJob.class);
-
-    String hadoopTokenFileLocation = System.getenv(HADOOP_TOKEN_FILE_LOCATION);
-    if (hadoopTokenFileLocation != null) {
-      job.getConfiguration().set(MAPREDUCE_JOB_CREDENTIALS_BINARY, hadoopTokenFileLocation);
-    }
+    setValidationConfigs(job, sampleAvroPath);
+    setHadoopJobConfigs(job, numInputPaths);
 
     // Avro Schema configs.
-    Schema avroSchema = getSchema(inputDataPath.get(0));
+    Schema avroSchema = getSchema(sampleAvroPath);
     _logger.info("Schema is: {}", avroSchema.toString(true));
-
-    // Validates configs against schema.
-    validateConfigsAgainstSchema(avroSchema);
-
-    // Mapper configs.
-    job.setMapperClass(SegmentPreprocessingMapper.class);
-    job.setMapOutputKeyClass(AvroKey.class);
-    job.setMapOutputValueClass(AvroValue.class);
-    job.getConfiguration().setInt(JobContext.NUM_MAPS, inputDataPath.size());
-
-    // Reducer configs.
-    job.setReducerClass(SegmentPreprocessingReducer.class);
-    job.setOutputKeyClass(AvroKey.class);
-    job.setOutputValueClass(NullWritable.class);
-
-    AvroMultipleOutputs.addNamedOutput(job, "avro", AvroKeyOutputFormat.class, avroSchema);
-    AvroMultipleOutputs.setCountersEnabled(job, true);
-    // Use LazyOutputFormat to avoid creating empty files.
-    LazyOutputFormat.setOutputFormatClass(job, AvroKeyOutputFormat.class);
-
-    // Input and output paths.
-    FileInputFormat.setInputPaths(job, _inputSegmentDir);
-    FileOutputFormat.setOutputPath(job, _preprocessedOutputDir);
-    _logger.info("Total number of files to be pre-processed: {}", inputDataPath.size());
+    setSchemaParams(job, avroSchema, numInputPaths);
 
     // Set up mapper output key
     Set<Schema.Field> fieldSet = new HashSet<>();
 
+    fetchPartitioningConfig();
+    fetchSortingConfig();
+    fetchResizingConfig();
+    validateConfigsAgainstSchema(avroSchema);
+
     // Partition configs.
-    int numReduceTasks = (_numberOfPartitions != 0) ? _numberOfPartitions : inputDataPath.size();
+    int numReduceTasks = (_numberOfPartitions != 0) ? _numberOfPartitions : inputDataPaths.size();
```

Comment:
Move the value assignment into the if clause because they are based on the same thing?